src/map.cpp

```diff
@@ -4672,6 +4672,28 @@ static void process_vehicle_items( vehicle *cur_veh, int part )
             apply_in_fridge(n);
         }
     }
+
+    const bool washmachine_here = cur_veh->part_flag( part, VPFLAG_WASHING_MACHINE ) && cur_veh->has_part( "WASHING_MACHINE", true );
+    bool washing_machine_finished = false;
+    if( washmachine_here ) {
+        for( auto &n : cur_veh->get_items( part ) ) {
+            const int washing_time = MINUTES( 90 );
+            int time_left = washing_time - calendar::turn.get_turn() + n.bday;
+            static const std::string filthy( "FILTHY" );
+            if( time_left == 0 ) {
+                n.item_tags.erase( filthy );
+                washing_machine_finished = true;
+                cur_veh->parts[part].enabled = false;
+            } else if( calendar::once_every( MINUTES( 15 ) ) ) {
+                add_msg( _("It should take %d minutes to finish washing."), time_left / MINUTES( 1 ) + 1 );
```

Comment:
This one could use the vehicle name too.
Consider a base with 3 washing machines on wheels. It's unlikely to happen (that detergent cost isn't low), but it's better to inform the player too much than too little, as long as it doesn't require more messages.